docs/data/data-grid/recipes-row-grouping/RowGroupingToolbar.tsx

```diff
@@ -0,0 +1,190 @@
+import * as React from 'react';
+import {
+  DataGridPremium,
+  Toolbar,
+  ToolbarButton,
+  useGridSelector,
+  useGridApiContext,
+  gridColumnLookupSelector,
+  gridColumnReorderDragColSelector,
+  gridRowGroupingSanitizedModelSelector,
+  useGridApiRef,
+} from '@mui/x-data-grid-premium';
+import { useDemoData } from '@mui/x-data-grid-generator';
+import Typography from '@mui/material/Typography';
+import Chip from '@mui/material/Chip';
+import ChevronRightIcon from '@mui/icons-material/ChevronRight';
+import MoveUpIcon from '@mui/icons-material/MoveUp';
+import { styled } from '@mui/material/styles';
+
+const StyledToolbar = styled(Toolbar)(({ theme }) => ({
+  justifyContent: 'flex-start',
+  overflow: 'auto',
+  padding: theme.spacing(1, 1.5),
+  gap: theme.spacing(1),
+}));
+
+function CustomToolbar() {
+  const apiRef = useGridApiContext();
+  const rowGroupingModel = useGridSelector(
+    apiRef,
+    gridRowGroupingSanitizedModelSelector,
+  );
+  const columnsLookup = useGridSelector(apiRef, gridColumnLookupSelector);
+  const draggedColumn = useGridSelector(apiRef, gridColumnReorderDragColSelector);
+  const [draggedChip, setDraggedChip] = React.useState<string | null>(null);
+
+  const handleToolbarDragOver = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+
+    const draggedField = draggedChip || draggedColumn;
+    if (draggedField && !rowGroupingModel.includes(draggedField)) {
```

Comment:
It would be great not to allow dropping non-groupable columns into the toolbar.
And disable chips for non-groupable columns that have been included in the grouping model programatically.

Reply:
> It would be great not to allow dropping non-groupable columns into the toolbar.

Makes sense 👍 

> And disable chips for non-groupable columns that have been included in the grouping model programatically.

Is there a use case for this? 🤔

Reply:
Yes, something like this: https://stackblitz.com/edit/1yktkhhd?file=src%2FDemo.tsx
In this demo, the rows are grouped by company, and you cannot ungroup them. You can add more grouping criteria, though.